com.incquerylabs.magicdraw.plugin.example/build.gradle

```diff
@@ -1,252 +1,9 @@
-            pluginLibraries += '\t\t<library name=\"' + file.absolutePath.replaceAll(~/.*build\/generated-dist\/plugins\/${groupName}\//, '') + '\"/>' + System.lineSeparator()
-        }
-        project.copy {
-            from 'src/main/dist/template/plugins/${group}/plugin.xml'
-            filter { String line -> line.replace('<!-- START AUTO-GENERATED -->', '<!-- START AUTO-GENERATED -->' + System.lineSeparator() + System.lineSeparator() + pluginLibraries) }
-            filter { String line -> line.replace('${human.version}', humanVersion) }
-            filter { String line -> line.replace('${internal.version}', internalVersion) }
-            into "build/generated-dist/plugins/${groupName}"
-        }
-
-        String descriptorFiles = ''
-        project.fileTree(dir: 'build/generated-dist').each { file ->
-            String relativePath = file.absolutePath.replaceAll(/.*build\/generated-dist\//, '')
-            descriptorFiles += '\t\t<file from=\"' + relativePath + '\" to=\"' + relativePath + '\"/>' + System.lineSeparator()
-        }
-        project.copy {
-            from "src/main/dist/template/data/resourcemanager/${descriptor}"
-            filter { String line -> line.replace('<!-- START AUTO-GENERATED -->', '<!-- START AUTO-GENERATED -->' + System.lineSeparator() + System.lineSeparator() + descriptorFiles) }
-            filter { String line -> line.replace('${human.version}', humanVersion) }
-            filter { String line -> line.replace('${internal.version}', internalVersion) }
-            filter { String line -> line.replace('${build.timestamp}', buildTimestamp) }
-            into 'build/generated-dist/data/resourcemanager'
-        }
-    }
-}
-
-task preDist(type: PreDist) {
-    humanVersion = version
-    internalVersion = buildNumber
-    groupName = magicdDrawGroupName
-    buildTimestamp = timestamp
-    descriptor = descriptorFile
-}
-preDist.dependsOn jar
-preDist.dependsOn javadocZip
-
-distributions {
-    main {
-        contents {
-            exclude 'template'
-            from 'build/generated-dist'
-        }
-    }
-}
-
-distZip {
-    includeEmptyDirs false
-    duplicatesStrategy DuplicatesStrategy.EXCLUDE
-    eachFile { file ->
-        file.setPath(path.substring(path.indexOf("/") + 1, path.length()))
-    }
-}
-distZip.dependsOn preDist
-
-installDist {
-    destinationDir = file('build/plugin')
-    doLast {
-        copy {
-            fileMode 0755
-            from 'build/dependency-cache/extracted'
-            from 'build/plugin'
-            into 'build/install'
-        }
-    }
-}
-installDist.dependsOn distZip
-
-def os = org.gradle.internal.os.OperatingSystem.current()
-def platformConfigSetting = 
-        (os.isWindows())
-		? ['-DWINCONFIG=true']
-		: ((os.isMacOsX())
-			? ['-DWINCONFIG=false', '-Xdock:name=MagicDraw', '-Xdock:icon=bin/md.icns', '-Dapple.laf.useScreenMenuBar=true']
-: ['-DWINCONFIG=false'])
-
 tasks.withType(JavaExec) {
```

Comment:
This part is common in both plugins, please extract it to the root build script.